### PR TITLE
Add rpfbagr tool

### PR DESCRIPTION
**Pull request by guillaume-gricourt**
FOR CONTRIBUTOR:
* [ ] - I have read the [CONTRIBUTING.md](https://github.com/galaxyproject/tools-iuc/blob/master/CONTRIBUTING.md) document and this tool is appropriate for the tools-iuc repo.
* [ ] - License permits unrestricted use (educational + commercial)
* [ ] - This PR adds a new tool or tool collection
* [ ] - This PR updates an existing tool or tool collection
* [ ] - This PR does something else (explain below)

### CODE REVIEW

File: tools/rp_fba_analysis/rp_fba_analysis.xml [hunk 1]
```diff
@@ -0,0 +1,135 @@
...
+                #end if
+            #end if
+            --max-knockouts '$max_knockouts'
+            --log-level ERROR
+            --output-file '$output_csv'
+    ]]></command>
+    <inputs>
+        <param name="model" type="data" format="sbml" label="Chassis model file" />
+        <param name="pathway" type="data" format="sbml" label="Pathway file" optional="True" />
+        <param name="medium" type="data" format="csv" label="Medium file" optional="True" />
```
[the remark this answers is not in the document]
Comment (guillaume-gricourt) [reply tools/rp_fba_analysis/rp_fba_analysis.xml:38]: Yes. Can I plan this in the next release ?

File: tools/rp_fba_analysis/rp_fba_analysis.xml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        <conditional name="input_strategy">
+            <param name="strategy" type="select" label="Strategy">
+                <option value="ko" selected="True">Gene knockouts</option>
+                <option value="ou" >Reaction knockouts</option>
+            </param>
+            <when value="ko">
+                <param name="substrate_rxn_id" type="text" label="Substrate reaction ID" value="" >
+                    <validator type="empty_field" message="Substrate reaction ID is required"/>
+                </param>
+                <param name="email" type="text" label="Email" value=""/>
```
[the remark this answers is not in the document]
Comment (guillaume-gricourt) [reply tools/rp_fba_analysis/rp_fba_analysis.xml:55]: The email is only used by the library biopython, module Entrez, to identify the user by the NCBI website. The name of the gene is obtained from its id with this optional feature.

File: tools/rp_fba_analysis/rp_fba_analysis.xml [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+            </when>
+            <when value="ou">
+            </when>
+        </conditional>
+        <param name="max_knockouts" type="integer" value="3" min="1" max="10" label="Number of items deleted to consider" />
+    </inputs>
+    <outputs>
+        <data name="output_csv" format="csv" label="${tool.name}(${input_strategy.strategy})" />
```
[the remark this answers is not in the document]
Comment (guillaume-gricourt) [reply tools/rp_fba_analysis/rp_fba_analysis.xml:63]: Yes. Can I plan this in the next release ?

**Comment by guillaume-gricourt** [comment]
> Thanks, @guillaume-gricourt! Very good quality. I have just a few questions.

Thank you so much. Tell me, if I need to add more documentations or other modifications.

File: tools/rp_fba_analysis/rp_fba_analysis.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (bgruening) [reply tools/rp_fba_analysis/rp_fba_analysis.xml:55]: Upps this is tricky. We don't want to have traces in the logs and as part of the provenance information that Galaxy saves - GDPR etc...

You could use the system email address inside an ENV variable if this is really needed. But then the users needs to have the same Galaxy email the NCBI email - which is a valid assumption?

File: tools/rp_fba_analysis/rp_fba_analysis.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (bgruening) [reply tools/rp_fba_analysis/rp_fba_analysis.xml:38]: Think so. Just keep in mind that Galaxy has million of tabular tools but only a handful of tools that can operate natively on CSV.

**Comment by bgruening** [comment]
> > Thanks, @guillaume-gricourt! Very good quality. I have just a few questions.
> 
> Thank you so much. Tell me, if I need to add more documentations or other modifications.

The test is failing, we need to get this green again :)

File: tools/rp_fba_analysis/rp_fba_analysis.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (guillaume-gricourt) [reply tools/rp_fba_analysis/rp_fba_analysis.xml:55]: I saw in environment variables : USER, USERNAME, LOGNAME when the runner was lunched, but nothing about EMAIL.
I understand these constraints and this feature is not essential. So I suggest, do not implement it. But I'm interested to known how you have thought to get the email in a ENV variable.

File: tools/rp_fba_analysis/rp_fba_analysis.xml [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+            --log-level ERROR
+            --output-file '$output_csv'
+    ]]></command>
+    <inputs>
+        <param name="model" type="data" format="sbml" label="Chassis model file" />
+        <param name="pathway" type="data" format="sbml" label="Pathway file" optional="True" />
+        <param name="medium" type="data" format="csv" label="Medium file" optional="True" />
+
+        <param name="biomass_rxn_id" type="text" label="Biomass reaction ID" value="" >
+            <validator type="empty_field" message="Biomass reaction ID is required"/>
```
[the remark this answers is not in the document]
Comment (guillaume-gricourt) [reply tools/rp_fba_analysis/rp_fba_analysis.xml:41]: Oh, you are right. I inspected the SBML file format and the ID label respects the ID of XML file format, so I will push a new version with a more constraint validator.

File: tools/rp_fba_analysis/rp_fba_analysis.xml (hunk 2 above)
[the remark this answers is not in the document]
Comment (bgruening) [reply tools/rp_fba_analysis/rp_fba_analysis.xml:55]: Try this here: https://github.com/galaxyproject/tools-iuc/blob/master/tools/ncbi_entrez_eutils/macros.xml#L8

File: tools/rpfbagr/rpfbagr.xml [hunk 5]
```diff
@@ -0,0 +1,153 @@
...
+* **Annotate**\: Add the gene's name thanks to the NCBI service.
+* **Time out**\: Restrict time of the analyse (minutes).
+
+Project Links
+---------------------
+
+* `rpfbagr <https://github.com/brsynth/rpfbagr>`_
+* `cameo <https://github.com/biosustain/cameo>`_
+
+Authors
```
Comment (bgruening) [inline tools/rpfbagr/rpfbagr.xml:136]: @guillaume-gricourt what about using this here: https://docs.galaxyproject.org/en/latest/dev/schema.html#tool-creator

File: tools/rpfbagr/rpfbagr.xml [hunk 6]
```diff
@@ -0,0 +1,153 @@
...
+            </when>
+        </conditional>
+    </inputs>
+    <outputs>
+        <data name="output_tsv" format="tsv" label="${tool.name}(${input_strategy.strategy})" />
+    </outputs>
+    <tests>
+        <test>
+        <!-- test 1: check if identical outputs are produced with R_BIOMASS__3 biomass name ID and BiGG model (iCN718)  -->
+            <param name="model" value="iAF1260.xml.gz" />
```
Comment (bgruening) [inline tools/rpfbagr/rpfbagr.xml:84]: maybe add here ftype="smbl"?

File: tools/rpfbagr/rpfbagr.xml (hunk 5 above)
Comment (guillaume-gricourt) [reply tools/rpfbagr/rpfbagr.xml]: You are right, it's amended #4209ab7

File: tools/rp_fba_analysis/rp_fba_analysis.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (guillaume-gricourt) [reply tools/rp_fba_analysis/rp_fba_analysis.xml]: Added ce17faf4b1bfb69293f01eaaca432d82fd22cd9d

File: tools/rp_fba_analysis/rp_fba_analysis.xml (hunk 1 above)
[the remark this answers is not in the document]
Comment (guillaume-gricourt) [reply tools/rp_fba_analysis/rp_fba_analysis.xml]: Added ce17faf4b1bfb69293f01eaaca432d82fd22cd9d

File: tools/rp_fba_analysis/rp_fba_analysis.xml (hunk 4 above)
[the remark this answers is not in the document]
Comment (guillaume-gricourt) [reply tools/rp_fba_analysis/rp_fba_analysis.xml]: Added 160c6eb9167e2cfddd55905e779746b1939c3b11

File: tools/rpfbagr/rpfbagr.xml (hunk 6 above)
Comment (guillaume-gricourt) [reply tools/rpfbagr/rpfbagr.xml]: Added 66c34a10c867f4a39cdbe39461d5cbcae892df8b

File: tools/rpfbagr/rpfbagr.xml [hunk 7]
```diff
@@ -24,14 +24,12 @@
...
                 --substrate-rxn-id '$input_strategy.substrate_rxn_id'
-                #if $input_strategy.annotate:
-                    --email "$__user_email__"
-                #end if
             #end if
             #if str($max_time_cond.max_time_param) == "yes":
                 --max-time '$max_time_cond.max_time'
             #end if
             --max-knockouts '$max_knockouts'
+            --email "$__user_email__"
```
Comment (bgruening) [inline tools/rpfbagr/rpfbagr.xml]: Should we add in the help section that the galaxy email will be used to query the NCBI database or should we add a Boolean like... Please agree to use your email for...

Comment (guillaume-gricourt) [reply tools/rpfbagr/rpfbagr.xml]: Yes, I will update the `help` section.

**Comment by guillaume-gricourt** [comment]
What do you think about these lines in the wrapper ?
```suggestion
    <command detect_errors="exit_code"><![CDATA[
         ## lib cobrapy writes in this directory
         export XDG_CACHE_HOME="\${_GALAXY_JOB_TMP_DIR:-.}";

         ## lib cameo needs this var set
         export USERNAME="$__user_email__";
```
The variables `_GALAXY_JOB_TMP_DIR` and `__user_email__` are appropriated ?

**Pull request closed without merging**